compiler/rustc_session/src/options.rs

```diff
@@ -1452,9 +1452,9 @@ options! {
     fewer_names: Option<bool> = (None, parse_opt_bool, [TRACKED],
         "reduce memory use by retaining fewer names within compilation artifacts (LLVM-IR) \
         (default: no)"),
-    flatten_format_args: bool = (false, parse_bool, [TRACKED],
+    flatten_format_args: bool = (true, parse_bool, [TRACKED],
```

Comment:
What's the plan for deleting the flag? A few weeks or releases?

Reply:
Is it a problem to just keep the flag around (permanently unstable)? I imagine it might be useful in the future for debugging when working on rustc.

Reply:
I'd prefer deleting flags that aren't needed. But if you think that it's useful to keep it we can keep it of course. Although I'm not sure whether it really is, we don't have a flag for each if after all :D

Reply:
I think it'll remain useful for as long as we keep improving/changing format_args/fmt::Arguments. If there's ever a bug report that we suspect might be related to this optimization, it seems very useful to be able to quickly check with -Zflatten-format-args=no to rule it out.